src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -859,8 +876,36 @@ def update_agents(cmd, client, resource_group_name, cluster_name, https_proxy=""
     # Setting the config dataplane endpoint
     config_dp_endpoint = get_config_dp_endpoint(cmd, connected_cluster.location)
 
+    if auto_upgrade != "false":
```

Comment:
What is the purpose of this check.? In the helm charts, it will always be present as either "true" or "false"[enum]

Reply:
this should be auto_upgrade == None